node/impl/full/sync.go

```diff
@@ -126,3 +126,29 @@ func (a *SyncAPI) SyncCheckBad(ctx context.Context, bcid cid.Cid) (string, error
 
 	return reason, nil
 }
+
+func (a *SyncAPI) SyncValidateTipset(ctx context.Context, tsk types.TipSetKey) (bool, error) {
+	ts, err := a.Syncer.ChainStore().LoadTipSet(tsk)
+	if err != nil {
+		return false, err
+	}
+
+	fts, err := a.Syncer.ChainStore().TryFillTipSet(ts)
+	if err != nil {
+		return false, err
+	}
+
+	for _, blk := range tsk.Cids() {
+		err = a.Syncer.ChainStore().UnmarkBlockAsValidated(ctx, blk)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	err = a.Syncer.ValidateTipSet(ctx, fts)
```

Comment:
I'd feel a bit better if instead of `UnmarkBlockAsValidated` we had some context value which we'd check in syncer.store.IsBlockValidated to disable the validation cache, especially given that this method has the `read` perm set

Reply:
Yeah, a 'force tipset revalidation' flow makes sense to me, rather than having it be a two step thing